### PR TITLE
 User can create account

### PR DESCRIPTION
**Pull request by EevanR**
[PT link](https://www.pivotaltracker.com/n/projects/2423721)

User can create account
Happy path
   navigates to sign up page
   fill in sign up form
   successfully creates an account and is greeted with success message

### CODE REVIEW

File: features/user_can_create_account.feature [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Feature: User can create account
+    As a user
+    In order to use the service
+    I need to be able to create an account
+
+    Background:
+        Given I am on the landing page
+        And I click on "Sign up"
+        Then I should be on the Sign up page
```
Comment (oliverochman) [inline features/user_can_create_account.feature]: We never want to have any assertions in the background.

This test is also redundant. If we can fill out the form and successfully create an account, then we are on the sign up page. Please remove this test

File: features/step_definitions/basic_steps.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  
+And("I click on {string}") do |element|
+    click_on element
+end
+
+When("I fill in {string} with {string}") do |input, content|
+    fill_in input, with: content
+end
+
+Then("I should be on the Sign up page") do
```
Comment (oliverochman) [inline features/step_definitions/basic_steps.rb]: some of these steps need to be in an assertion step file

File: features/user_can_create_account.feature [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    Scenario: Successfully creates an account
+        When I fill in "Name" with "Bob"
+        And I fill in "Email" with "bob@bob.bob"
+        And I fill in "Password" with "bobbobbob"
+        And I fill in "Password confirmation" with "bobbobbob"
+        And I click on "Create"
+        Then I should be on the landing page
+        And I should see "Welcome! You have signed up successfully."
+        And I should see "Hello, Bob"
```
Comment (oliverochman) [inline features/user_can_create_account.feature]: Please go ahead and test the multiple sad paths in this file as well.

File: features/user_can_create_account.feature (hunk 1 above)
Comment (mantanquzzmo) [reply features/user_can_create_account.feature]: Yes, agree. Was added for debugging the little bomb.

File: features/user_can_create_account.feature (hunk 3 above)
Comment (mantanquzzmo) [reply features/user_can_create_account.feature]: Yes. But will split it into more PRs.

File: features/step_definitions/basic_steps.rb (hunk 2 above)
Comment (mantanquzzmo) [reply features/step_definitions/basic_steps.rb]: ok

File: features/step_definitions/assertians.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+Then("I should be on the landing page") do
+  expect(current_path).to eq root_path
+end
+
+Then("I should see {string}") do |string|
```
Comment (oliverochman) [inline features/step_definitions/assertians.rb]: Please change the variable name from `string` to `content`. More descriptive.

File: features/user_can_create_account.feature (hunk 3 above)
Comment (oliverochman) [reply features/user_can_create_account.feature]: Nope nope, we will have the sad paths in this feature file, in this PR. 1 PT entry = 1 PR

**Pull request merged**